cmd2/cmd2.py

```diff
@@ -381,6 +399,218 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, *,
+        setattr(self, help_func_name, cmd_help)
+
+    def uninstall_command_set(self, cmdset: CommandSet):
+        """
+        Uninstalls a CommandSet and unloads all associated commands
+        :param cmdset: CommandSet to uninstall
+        """
+        if cmdset in self._installed_command_sets:
+
+            self._unregister_subcommands(cmdset)
+
+            methods = inspect.getmembers(
+                cmdset,
+                predicate=lambda meth: inspect.ismethod(meth) and meth.__name__.startswith(COMMAND_FUNC_PREFIX))
+
+            for method in methods:
+                cmd_name = method[0][len(COMMAND_FUNC_PREFIX):]
+
+                delattr(self, COMMAND_FUNC_PREFIX + cmd_name)
+
+                if hasattr(self, COMPLETER_FUNC_PREFIX + cmd_name):
+                    delattr(self, COMPLETER_FUNC_PREFIX + cmd_name)
+                if hasattr(self, HELP_FUNC_PREFIX + cmd_name):
+                    delattr(self, HELP_FUNC_PREFIX + cmd_name)
+
+            cmdset.on_unregister(self)
+            self._installed_command_sets.remove(cmdset)
+
+    def _register_subcommands(self, cmdset: Union[CommandSet, 'Cmd']) -> None:
+        """
+        Register sub-commands with their base command
+
+        :param cmdset: CommandSet containing sub-commands
+        """
+        if not (cmdset is self or cmdset in self._installed_command_sets):
+            raise ValueError('Adding sub-commands from an unregistered CommandSet')
+
+        # find all methods that start with the sub-command prefix
+        methods = inspect.getmembers(
+            cmdset,
+            predicate=lambda meth: (inspect.ismethod(meth) or isinstance(meth, Callable))
+            and hasattr(meth, constants.SUBCMD_ATTR_NAME)
+            and hasattr(meth, constants.SUBCMD_ATTR_COMMAND)
+            and hasattr(meth, constants.CMD_ATTR_ARGPARSER)
+        )
+
+        # iterate through all matching methods
+        for method_name, method in methods:
+            subcommand_name = getattr(method, constants.SUBCMD_ATTR_NAME)
+            command_name = getattr(method, constants.SUBCMD_ATTR_COMMAND)
+            subcmd_parser = getattr(method, constants.CMD_ATTR_ARGPARSER)
+
+            # Search for the base command function and verify it has an argparser defined
+            command_func = self.cmd_func(command_name)
+            if command_func is None or not hasattr(command_func, constants.CMD_ATTR_ARGPARSER):
+                raise TypeError('Could not find command "{}" needed by sub-command: {}'
+                                .format(command_name, str(method)))
+            command_parser = getattr(command_func, constants.CMD_ATTR_ARGPARSER)
+            if command_parser is None:
+                raise TypeError('Could not find argparser for command "{}" needed by sub-command: {}'
+                                .format(command_name, str(method)))
+
+            if hasattr(method, '__doc__') and method.__doc__ is not None:
```

Comment:
The `as_subcommand_to` decorator should take a `help` argument instead of using the methods docstring.